app/models/collection_item.rb

```diff
@@ -191,12 +191,15 @@ def send_work_invitation
 
   after_update :notify_of_status_change
   def notify_of_status_change
-    if saved_change_to_unrevealed? && item.respond_to?(:new_recipients)
-      # making sure notify_recipients in the work model has not already notified
-      if item.new_recipients.present?
-        notify_of_reveal
-      end
-    end
+    unrevealed_collection_ids = item
+      .collection_items
+      .unrevealed
+      .approved_by_user
+      .pluck(:collection_id)
+    # If the item is also in other unrevealed collections, skip notifications.
+    return if (unrevealed_collection_ids - [collection_id]).any?
```

Comment:
FYI, the extra complexity here is unnecessary. You don't need to worry about whether the current `CollectionItem` is set wrong, because `after_update` hooks are run inside the same transaction that set that value, which means that you're guaranteed to see the updates from that transaction. You should just be able to copy the check from `lib/collectible.rb`:
https://github.com/otwcode/otwarchive/blob/c364d7f1a76af0d73a260158bd824e1c52962637/lib/collectible.rb#L118

(You can't use the new value of `item.unrevealed` or `work.unrevealed` here because `in_unrevealed_collection` is updated in the `after_save` hook `update_work`, and `after_save` hooks are always run after `after_update` hooks. But the `CollectionItem`s can't run into stale data issues here.)

Also, as I mentioned, this approach can still result in seeing notifications for unrevealed gifts if the transaction is rolled back (hence my suggestion about `after_commit_everywhere`). But I believe that's a pre-existing bug, so I guess it's okay to leave it in.

Reply:
I was going to update this now that we've added the gem, but it looks like #3818 will take care of using `after_commit_everywhere` for `notify_on_reveal`, correct?

Reply:
Yep, that should take care of it.